src/database.c

```diff
@@ -184,6 +187,29 @@ static void abort_transaction(MDB_txn *txn)
 	dbi_init = 0;
 }
 
+/*
+ * Convert path to a hash value. Used when the path exceeds the LMDB key limit(511).
+ * Note: Returned value must be deallocated.
+ */
+char *path_to_hash(const char *path)
+{
+    gcry_md_hd_t h;
+    unsigned int len;
+    char *digest, *hptr;
+
+    gcry_md_open(&h, GCRY_MD_SHA256, GCRY_MD_FLAG_SECURE); /* initialize the hash context */
+    gcry_md_write(h, path, strlen(path)); /* hash the text */
+
+    hptr = (char *)gcry_md_read(h, GCRY_MD_SHA256);
+
+    len = gcry_md_get_algo_dlen(GCRY_MD_SHA256) * sizeof(char);
+    digest = malloc((2 * len) + 1);
```

Comment:
Consider accommodating for an error path post calling `malloc` and perhaps any other routines called from here? Not entirely sure how it would affect the logic in the caller if NULL was returned for example... Also, calling `bytes2hex()` with a NULL value might be fun...